django/db/migrations/serializer.py

```diff
@@ -0,0 +1,387 @@
+    def _format(self):
+        return "frozenset([%s])"
+
+
+class FunctionTypeSerializer(BaseSerializer):
+    def serialize(self):
+        if getattr(self.value, "__self__", None) and isinstance(self.value.__self__, type):
+            klass = self.value.__self__
+            module = klass.__module__
+            return "%s.%s.%s" % (module, klass.__name__, self.value.__name__), {"import %s" % module}
+        # Further error checking
+        if self.value.__name__ == '<lambda>':
+            raise ValueError("Cannot serialize function: lambda")
+        if self.value.__module__ is None:
+            raise ValueError("Cannot serialize function %r: No module" % self.value)
+        # Python 3 is a lot easier, and only uses this branch if it's not local.
+        if getattr(self.value, "__qualname__", None) and getattr(self.value, "__module__", None):
+            if "<" not in self.value.__qualname__:  # Qualname can include <locals>
+                return "%s.%s" % \
+                    (self.value.__module__, self.value.__qualname__), {"import %s" % self.value.__module__}
+        # Python 2/fallback version
+        module_name = self.value.__module__
+        # Make sure it's actually there and not an unbound method
+        module = import_module(module_name)
+        if not hasattr(module, self.value.__name__):
+            raise ValueError(
+                "Could not find function %s in %s.\n"
+                "Please note that due to Python 2 limitations, you cannot "
+                "serialize unbound method functions (e.g. a method "
+                "declared and used in the same class body). Please move "
+                "the function into the main module body to use migrations.\n"
+                "For more information, see "
+                "https://docs.djangoproject.com/en/%s/topics/migrations/#serializing-values"
+                % (self.value.__name__, module_name, get_docs_version()))
+        # Needed on Python 2 only
+        if module_name == '__builtin__':
+            return self.value.__name__, set()
+        return "%s.%s" % (module_name, self.value.__name__), {"import %s" % module_name}
+
+
+class FunctoolsPartialSerializer(BaseSerializer):
+    def serialize(self):
+        imports = {'import functools'}
+        # Serialize functools.partial() arguments
+        func_string, func_imports = serializer_factory(self.value.func).serialize()
+        args_string, args_imports = serializer_factory(self.value.args).serialize()
+        keywords_string, keywords_imports = serializer_factory(self.value.keywords).serialize()
+        # Add any imports needed by arguments
+        imports.update(func_imports)
+        imports.update(args_imports)
+        imports.update(keywords_imports)
+        return (
+            "functools.partial(%s, *%s, **%s)" % (
+                func_string, args_string, keywords_string,
+            ),
+            imports,
+        )
+
+
+class IterableSerializer(BaseSerializer):
```

Comment:
It's very similar, but it uses len(strings) instead of len(self.value). We could have a property or method called length_to_serialize() that returns strings for Iterables or self.value for Tuples, but I wasn't sure it was worthwhile. 

Do you think I should add that in IterableSerializer and have TupleSerializer override it?